### PR TITLE
add io.mpv.Mpv

### PR DESCRIPTION
**Pull request by paulcarroty**
https://mpv.io/

### CODE REVIEW

**Comment by gasinvein** [comment]
I believe the ID should be `io.mpv.Mpv`. Currently, it suggest that `github.com` domain is owned by Mpv.

**Comment by LeandroStanger** [comment]
bot, build io.mpv.Mpv

File: io.mpv.Mpv.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
+{
+  "app-id": "io.github.Mpv",
```
Comment (AsciiWolf) [inline io.mpv.Mpv.json]: The appid needs to be changed here as well.

**Comment by LeandroStanger** [comment]
bot, build io.mpv.Mpv

File: io.mpv.Mpv.json [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/mpv-player/mpv/archive/v0.29.1.tar.gz",
+          "sha256": "f9f9d461d1990f9728660b4ccb0e8cb5dce29ccaa6af567bec481b79291ca623"
+        },
+        {
+          "type": "file",
+          "url": "https://waf.io/waf-2.0.12",
```
Comment (Erick555) [inline io.mpv.Mpv.json]: Use https://waf.io/waf-2.0.17

File: io.mpv.Mpv.json [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            "/lib/*/*.la",
+            "/lib*/*/*/*.la",
+            "/lib/pkgconfig",
+            "/share/man"
+          ],
+          "config-opts": ["--disable-static", "--disable-bpf"],
+          "sources": [
+            {
+              "type": "archive",
+              "url": "https://linuxtv.org/downloads/v4l-utils/v4l-utils-1.16.2.tar.bz2",
```
Comment (Erick555) [inline io.mpv.Mpv.json]: Use https://linuxtv.org/downloads/v4l-utils/v4l-utils-1.16.6.tar.bz2

File: io.mpv.Mpv.json [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            "--disable-programs",
+            "--disable-encoders",
+            "--disable-muxers",
+            "--enable-encoder=png",
+            "--enable-libv4l2"
+          ],
+          "sources": [
+            {
+              "type": "archive",
+              "url": "https://ffmpeg.org/releases/ffmpeg-4.1.tar.xz",
```
Comment (Erick555) [inline io.mpv.Mpv.json]: Use https://ffmpeg.org/releases/ffmpeg-4.1.3.tar.xz

**Comment by nedrichards** [comment]
bot, build io.mpv.Mpv

**Comment by AsciiWolf** [comment]
bot, build io.mpv.Mpv

File: io.mpv.Mpv.json [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+      "buildsystem": "simple",
+      "config-opts": ["--buildtype=release"],
+      "cleanup": ["/include", "/lib/pkgconfig"],
+      "build-commands": [
+        "python3 waf configure --prefix=/app --disable-build-date --disable-oss-audio --disable-alsa --enable-tv",
+        "python3 waf build",
+        "python3 waf install"
+      ],
+      "post-install": [
+        "install -D -m644 io.mpv.Mpv.appdata.xml /app/share/appdata/io.mpv.Mpv.appdata.xml"
```
Comment (AsciiWolf) [inline io.mpv.Mpv.json]: You must add something like this to sources:
```
                {
                    "type": "file",
                    "path": "io.mpv.Mpv.appdata.xml"
                },
```

**Comment by AsciiWolf** [comment]
bot, build io.mpv.Mpv

File: io.mpv.Mpv.json [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+          "sha256": "f9f9d461d1990f9728660b4ccb0e8cb5dce29ccaa6af567bec481b79291ca623"
+        },
+        {
+          "type": "file",
+          "url": "https://waf.io/waf-2.0.17",
+          "sha256": "e05a7f0e81dc3af7e744ddf7c7a334e3504967108fe51fab45808015ba1c0de0",
+          "dest-filename": "waf"
+        },
+        {
+                    "type": "file",
```
Comment (AsciiWolf) [inline io.mpv.Mpv.json]: Looks good, just fix the indentation. :-)

File: io.mpv.Mpv.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      <li>ao_jack: Fix for mpv attempting to connect to non-standard devices such as MIDI.</li>
+      <li>ao_openal: Drop support for Apple's bundled OpenAL due to lack of features to successfully build.</li>
+      <li>ao_pulse: Fix for audio buffer length calculation.</li>
+      <li>demux_lavf: Fix EDL playback of certain DASH streams.</li>
+      <li>demux_lavf: Mark v4l streams as not seekable.</li>
+      <li>demux_raw: Fix for gapless playback from raw audio input.</li>
+      <li>drm_atomic: Fix to enable video output with devices without a video overlay plane.</li>
+      <li>encode: Fix for crashes in case of failure to write to output.</li>
+      <li>encode: Fix for encoding when utilizing lavfi-complex.</li>
+      <li>Fix to strip HDR peak metadata if attached to SDR video (<a class="issue-link js-issue-link" data-error-text="Failed to load issue title" data-id="357030303" data-permission-text="Issue title is private" data-url="https://github.com/mpv-player/mpv/issues/6111" data-hovercard-type="issue" data-hovercard-url="/mpv-player/mpv/issues/6111/hovercard" href="https://github.com/mpv-player/mpv/issues/6111">#6111</a>)</li>
```
Comment (AsciiWolf) [inline io.mpv.Mpv.appdata.xml]: Remove the a class links. They do not seem to be useful anyway ("Failed to load issue title").

Comment (AsciiWolf) [reply io.mpv.Mpv.appdata.xml]: By the way, just a small typo: There is a missing dot at the end of the sentence. :-)

File: io.mpv.Mpv.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      <li>demux_lavf: Mark v4l streams as not seekable.</li>
+      <li>demux_raw: Fix for gapless playback from raw audio input.</li>
+      <li>drm_atomic: Fix to enable video output with devices without a video overlay plane.</li>
+      <li>encode: Fix for crashes in case of failure to write to output.</li>
+      <li>encode: Fix for encoding when utilizing lavfi-complex.</li>
+      <li>Fix to strip HDR peak metadata if attached to SDR video.</li>
+      <li>macOS: Capability to fall back on SW rendering for testing purposes.</li>
+      <li>macOS: Fix for issues with the main menu.</li>
+      <li>macOS: Fix for side by side split view.</li>
+      <li>macOS: Fixes for crashes, most notably those happening on 10.14 .</li>
```
Comment (AsciiWolf) [inline io.mpv.Mpv.appdata.xml]: Trailing space before the dot, but that's just a small cosmetic thing. :-)

**Comment by AsciiWolf** [comment]
bot, build io.mpv.Mpv

**Comment by AsciiWolf** [comment]
bot, build io.mpv.Mpv

**Comment by AsciiWolf** [comment]
@paulcarroty Please check the AppData file with `appstream-util validate-relax` command and fix any issues found.

**Comment by AsciiWolf** [comment]
Ignore my previous comment. The issue seems to be caused by a desktop/icon file name, rename-desktop-file and rename-icon need to be added to the manifest json. See [File renaming](http://docs.flatpak.org/en/latest/manifests.html#file-renaming) in the Flatpak docs.

File: io.mpv.Mpv.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2015-2019 gnome-mpv -->
+<component type="desktop">
+ <id>io.mpv.Mpv</id>
+ <metadata_license>CC-BY-SA-4.0</metadata_license>
+ <project_license>GPLv2+ and LGPLv2+</project_license>
+ <name>mpv</name>
+ <summary>a free, open source, and cross-platform media player</summary>
+ <launchable type="desktop-id">io.mpv.Mpv</launchable>
```
Comment (TingPing) [inline io.mpv.Mpv.appdata.xml]:
```suggestion
 <launchable type="desktop-id">io.mpv.Mpv.desktop</launchable>
```

File: io.mpv.Mpv.appdata.xml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+ <translation type="gettext">mpv</translation>
+ <content_rating type="oars-1.1" />
+ <provides>
+  <id>io.mpv.Mpv.desktop</id>
```
Comment (TingPing) [inline io.mpv.Mpv.appdata.xml]: Just remove this, a previous package never existed.

File: io.mpv.Mpv.json [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+    "--share=network",
+    "--socket=pulseaudio",
+    "--filesystem=host",
+    "--env=LC_NUMERIC=C"
+  ],
+  "modules": [
+    {
+      "name": "mpv",
+      "buildsystem": "simple",
+      "config-opts": ["--buildtype=release"],
```
Comment (TingPing) [inline io.mpv.Mpv.json]:
```suggestion
```

This does nothing.

**Comment by ramcq** [comment]
bot, build io.mpv.Mpv

**Comment by nedrichards** [comment]
bot, build io.mpv.Mpv

**Comment by nedrichards** [comment]
What's the relation to upstream of this flatpak?

**Comment by paulcarroty** [comment]
Neutral, they prefer snap.

**Comment by AsciiWolf** [comment]
bot, build io.mpv.Mpv

**Comment by paulcarroty** [comment]
```
[📦 io.mpv.Mpv ~]$ mpv --gpu-api=help
GPU APIs (contexts):
    auto (autodetect)
    opengl (x11probe)
    opengl (x11egl)
    opengl (x11)
    opengl (wayland)
    opengl (drm)
    opengl (vdpauglx)
    vulkan (waylandvk)
    vulkan (x11vk)
```
 🏅 for @TingPing.

File: io.mpv.Mpv.appdata.xml [hunk 12]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline io.mpv.Mpv.appdata.xml]: Should be contributed upstream.

**Comment by LeandroStanger** [comment]
bot, build io.mpv.Mpv

**Comment by nedrichards** [comment]
OK then, so the only thing that stands in the way is an upstream PR pushing the appdata file - since that's generally useful for everyone on Linux, not just flatpak. Sad that upstream don't want to be involved, do you have a list of maintainers, just @paulcarroty? 

It'd be nice to send a PR to https://github.com/mpv-player/mpv.io as well once this is added to get on this page https://mpv.io/installation/

**Comment by TingPing** [comment]
I imagine the reason upstream prefers snap is because its primary used from the cli and flatpak requires a different command, making all existing docs unusable.

File: io.mpv.Mpv.appdata.xml [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+<!-- Copyright 2012-2019 mpv.io -->
+<component type="desktop">
```
Comment (nedrichards) [inline io.mpv.Mpv.appdata.xml]: this is just CLI, right? if so this should be console-application, right? https://www.freedesktop.org/software/appstream/docs/sect-Metadata-ConsoleApplication.html

Comment (paulcarroty) [reply io.mpv.Mpv.appdata.xml]: GUI mode with drag'n'drop exists.

File: io.mpv.Mpv.appdata.xml [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+ <id>io.mpv.Mpv</id>
+ <metadata_license>CC-BY-SA-4.0</metadata_license>
+ <project_license>GPL-2.0+ AND LGPL-2.0-or-later</project_license>
+ <name>mpv</name>
+ <summary>a free, open source, and cross-platform media player</summary>
+ <launchable type="desktop-id">io.mpv.Mpv.desktop</launchable>
```
Comment (AsciiWolf) [inline io.mpv.Mpv.appdata.xml]: Is the desktop file correctly renamed by flatpak-builder? It seems to be called `mpv.desktop` in upstream.

**Comment by nedrichards** [comment]
bot, build io.mpv.Mpv

File: io.mpv.Mpv.json [hunk 15]
```diff
@@ -0,0 +1,154 @@
+{
+  "app-id": "io.mpv.Mpv",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "18.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "mpv",
+  "rename-desktop-file": "mpv.desktop",
```
Comment (AsciiWolf) [inline io.mpv.Mpv.json]: `rename-icon` may be also needed, but I am not sure.

**Comment by AsciiWolf** [comment]
bot, build io.mpv.Mpv

File: io.mpv.Mpv.json [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+  "rename-icon":"mpv",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=wayland",
+    "--device=all",
+    "--share=network",
+    "--socket=pulseaudio",
+    "--filesystem=host",
```
Comment (gasinvein) [inline io.mpv.Mpv.json]: Does MPV ever need write access? If not,
```suggestion
    "--filesystem=host:ro",
```

Comment (paulcarroty) [reply io.mpv.Mpv.json]: It need write access for plugins.

Comment (gasinvein) [reply io.mpv.Mpv.json]: Plugins are stored in MPV config dir, which is mounted read-writable even with `host:ro`.

Comment (paulcarroty) [reply io.mpv.Mpv.json]: If you will test all subtitle addons and write the report, we can continue the discuss.

Comment (gasinvein) [reply io.mpv.Mpv.json]: Can you point to a specific plugin that writes to a directory outside xdg-config/xdg-data?

Comment (paulcarroty) [reply io.mpv.Mpv.json]: Guess all subtitle plugins allows to download subtitles to multimedia folder.

Comment (gasinvein) [reply io.mpv.Mpv.json]: Isn't it too specific use-case to expose whole filesystem for _all_ users? Users who actually need write access to host, may add it via overrides.

Comment (barthalion) [reply io.mpv.Mpv.json]: It'd be appreciated if you could stop resolving conversations that are still in progress.

**Comment by barthalion** [comment]
Have you submitted appdata upstream or you will pretend no one asked for that?

**Comment by paulcarroty** [comment]
https://github.com/flathub/flathub/pull/1057#issuecomment-508337423

**Comment by barthalion** [comment]
Appdata is not specific to flatpak.

**Comment by nedrichards** [comment]
@paulcarroty specifically an upstream PR https://github.com/flathub/flathub/pull/1057#issuecomment-508563561 it doesn't have to be merged but it does have to be offered.

**Comment by paulcarroty** [comment]
https://github.com/mpv-player/mpv/issues/2987

**Comment by gasinvein** [comment]
Where does the info about that MPV developers prefer snap comes from, btw? I didn't find neither official snap package, nor issues/PRs related to it.

**Comment by TingPing** [comment]
> [mpv-player/mpv#2987](https://github.com/mpv-player/mpv/issues/2987)

I think its worth trying again with more detail. In those 3 years every distros app store supports this standard now.

File: io.mpv.Mpv.json (hunk 16 above)
Comment (FakeShemp) [reply io.mpv.Mpv.json]: Screenshot needs write access, though I guess only to `xdg-pictures`.

**Comment by paulcarroty** [comment]
https://github.com/mpv-player/mpv/tree/appstream

**Comment by TingPing** [comment]
We've decided to not move forward with this at the moment.

You have a history of abandoning packages here and have not been the easiest to work with so we would prefer a different maintainer for this application.

I'll leave this open for a bit if anybody else wants to step up.

**Pull request closed without merging**